### PR TITLE
fix(core): manage device canvas context lifecycle

### PR DESCRIPTION
**Pull request by ibgreen-openai**
## Summary

- make `Device` own and tear down every `CanvasContext` and `PresentationContext` wrapper created through it
- add reference-counted release for reused and repeatedly attached WebGL devices
- add terminal `device.detach()` for exclusive owners that need to preserve and recover the backend handle
- document lifecycle, reuse, attach, detach, and upgrade behavior

## Motivation

This started from [deck.gl #10355](https://github.com/visgl/deck.gl/issues/10355) and the proposed deck-side cleanup in [deck.gl #10356](https://github.com/visgl/deck.gl/pull/10356), where dropping a `CanvasContext` without destroying it leaves resize, visibility, DPR, and position observers rooted after Deck finalization.

That leak is another instance in the long-running stream of canvas-related sync/async initialization, teardown, reuse, and attachment fixes across deck.gl and luma.gl. Rather than add another cleanup layer on top of the others, this PR moves as much of the lifecycle contract as possible to the root in luma.gl: the `Device` that creates context wrappers now owns their final teardown.

## Changes

- `Device` tracks managed canvas and presentation wrappers, destroys them in reverse creation order, and rejects new wrappers after final teardown.
- `CanvasSurface.destroy()` unregisters itself while preserving idempotent observer shutdown.
- WebGL, WebGPU, and Null devices register default, additional, and presentation wrappers and release backend-specific cached framebuffer/texture wrappers.
- Reused WebGL devices retain one logical reference per successful acquisition; final `destroy()` performs wrapper teardown and clears the WebGL context's luma metadata.
- Repeated `webgl2Adapter.attach()` calls also retain references, and attached handles are marked borrowed.
- `device.detach()` requires exclusive ownership, destroys luma wrappers, clears backend association state, preserves the native handle, and returns it.
- Owned WebGPU handles are destroyed only by final `destroy()`; `detach()` preserves them. WebGPU handle attachment remains intentionally unimplemented, but the ownership model is now defined for it.
- Added a dedicated Device lifecycle documentation section plus release and upgrade notes.

## Verification

- `nvm use`
- `yarn install`
- `yarn lint fix`
- `yarn build`
- `yarn test`
- `yarn website:build`
- `(cd website && yarn build)`
- pre-commit `test-fast`: 51 test files passed, 178 tests passed

### CODE REVIEW

File: modules/webgl/src/adapter/webgl-device.ts [hunk 1]
```diff
@@ -271,25 +274,34 @@ export class WebGLDevice extends Device {
...
+    if (!this._releaseDeviceReference()) {
+      return;
     }
+
+    this._finalizeDevice();
+  }
+
+  override detach(): WebGL2RenderingContext {
+    this._detachDeviceReference();
+    this._finalizeDevice();
```
[the remark this answers is not in the document]
Comment (ibgreen-openai) [reply modules/webgl/src/adapter/webgl-device.ts:295]: Fixed in `7723ea416`. `WebGLDevice` now retains its `WebGLStateTracker`, and final teardown calls `untrackState()`, which restores the original hooked WebGL methods and clears `gl.lumaState` before the context can be reattached. I added a detach/reattach regression test that verifies `gl.enable()` reaches the native setter exactly once.

File: modules/webgl/test/adapter/resources/webgl-render-pass.spec.ts [hunk 2]
```diff
@@ -8,6 +8,8 @@ import {GL} from '@luma.gl/webgl/constants';
 import {WEBGLRenderPass} from '@luma.gl/webgl';
 import {getWebGLTestDevice} from '@luma.gl/test-utils';
 
+// getWebGLTestDevice() returns a cached shared fixture; individual tests must not destroy it.
```
[the remark this answers is not in the document]
Comment (ibgreen-openai) [reply modules/webgl/test/adapter/resources/webgl-render-pass.spec.ts]: Fixed in `7723ea416`. Rather than silently recreating terminally destroyed cached fixtures, cached test devices now install non-writable `destroy()` / `detach()` guards that throw. I also removed the invalid teardown calls from the affected cached-fixture specs, including `webgl-render-pipeline.spec.ts`, and added a test for the guard.

File: modules/webgpu/src/adapter/webgpu-device.ts [hunk 3]
```diff
@@ -151,10 +153,27 @@ export class WebGPUDevice extends Device {
...
+    if (!this._releaseDeviceReference()) {
+      return;
+    }
+
+    this._finalizeDevice({destroyHandle: this._ownsHandle});
+  }
+
+  override detach(): GPUDevice {
+    this._detachDeviceReference();
+    this._finalizeDevice({destroyHandle: false});
```
[the remark this answers is not in the document]
Comment (ibgreen-openai) [reply modules/webgpu/src/adapter/webgpu-device.ts:169]: Fixed in `7723ea416`. WebGPU devices now store and remove the `uncapturederror` listener during finalization. The native loss callback captures a small loss-state object instead of `this`, so detach no longer keeps the luma wrapper alive while `device.lost` and `isLost` still update. I added detach coverage for both behaviors.

**Comment by ibgreen-openai** [comment]
Close temporarily until we focus on this again
